encoder.go

```diff
@@ -62,6 +62,11 @@ func (e *Encoder) encode(v reflect.Value, dst map[string][]string) error {
 			continue
 		}
 
+		if v.Field(i).Type().Kind() == reflect.Ptr && v.Field(i).Elem().IsValid() && v.Field(i).Elem().Type().Kind() == reflect.Struct {
```

Comment:
Can you add a comment here that examples the reflect method chaining? The reflect API is a little obtuse, and thus making it clear why this calls what it does will help future maintainers & contributors.

(I understand it, but not everyone does!)

Reply:
Done 👍